.github/workflows/nodejs.yml

```diff
@@ -195,6 +195,68 @@ jobs:
         if: always()
         run: docker stop rippled-service
 
+  generate-documentation:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [18.x]
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Setup npm version 10
+        run: |
+          npm i -g npm@10 --registry=https://registry.npmjs.org
+
+      - name: Cache node modules
+        id: cache-nodemodules
+        uses: actions/cache@v4
+        env:
+          cache-name: cache-node-modules
+        with:
+          # caching node_modules
+          path: |
+            node_modules
+            */*/node_modules
+          key: ${{ runner.os }}-deps-${{ matrix.node-version }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-deps-${{ matrix.node-version }}-
+
+      - name: Install Dependencies
+        if: steps.cache-nodemodules.outputs.cache-hit != 'true'
+        run: npm ci
+
+      - run: npm run build
+
+      - name: Generate Documentation
+        run: npm run docgen
+
+      - name: Upload documentation files as artifact
```

Comment:
It doesn't look like this PR actually commits the docs changes to the repo

Reply:
Yes, that is true.
> Additionally, the settings of this repository will be updated. Documentation will be deployed from Github Actions, instead of reading from the docs/ folder (this is the existing behavior)


This PR proposes a different way of deploying the documentation. `docs/` folder is obviated by the Github Actions workflow.